inputremapper/injection/mapping_handlers/rel_to_rel_handler.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# input-remapper - GUI for device specific keyboard mappings
+# Copyright (C) 2022 sezanzeb <proxima@sezanzeb.de>
+#
+# This file is part of input-remapper.
+#
+# input-remapper is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# input-remapper is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with input-remapper.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+from typing import Dict, Tuple, Optional
+
+import evdev
+from evdev.ecodes import (
+    EV_REL,
+    REL_X,
+    REL_Y,
+    REL_WHEEL,
+    REL_HWHEEL,
+    REL_WHEEL_HI_RES,
+    REL_HWHEEL_HI_RES,
+)
+
+from inputremapper import exceptions
+from inputremapper.configs.mapping import Mapping
+from inputremapper.event_combination import EventCombination
+from inputremapper.injection.global_uinputs import global_uinputs
+from inputremapper.injection.mapping_handlers.axis_transform import Transformation
+from inputremapper.injection.mapping_handlers.mapping_handler import (
+    MappingHandler,
+    HandlerEnums,
+    InputEventHandler,
+)
+from inputremapper.input_event import InputEvent
+from inputremapper.logger import logger
+
+
+def is_wheel(event) -> bool:
+    return event.type == EV_REL and event.code in (REL_WHEEL, REL_HWHEEL)
+
+
+def is_high_res_wheel(event) -> bool:
+    return event.type == EV_REL and event.code in (REL_WHEEL_HI_RES, REL_HWHEEL_HI_RES)
+
+
+class Remainder:
+    _scale: int
+    _remainder: float
+
+    def __init__(self, scale):
+        self._scale = scale
+        self._remainder = 0
+
+    def input(self, value):
+        # if the mouse moves very slow, it might not move at all because of the
+        # int-conversion (which is required when writing). store the remainder
+        # (the decimal places) and add it up, until the mouse moves a little.
+        scaled = value * self._scale
+        floored = int(scaled)
+        self._remainder += scaled - floored
+        if abs(self._remainder) >= 1:
+            output_value = int(scaled + self._remainder)
+            self._remainder = self._remainder - output_value
+        else:
+            output_value = floored
+
+        return output_value
```

Comment:
```suggestion
        scaled = value * self._scale + self._remainder
        self._remainder = math.fmod(scaled, 1)
        return int(scaled)
```
This seems to solve the performance issues with the re-2-rel handler.

Reply:
It's a 19.5% performance increase for me